src/store/postgres/event_store.rs

```diff
@@ -58,6 +58,7 @@ where
     pub(super) transactional_event_handlers:
         Vec<Box<dyn TransactionalEventHandler<A, PgStoreError, PgConnection> + Send>>,
     pub(super) event_buses: Vec<Box<dyn EventBus<A> + Send>>,
+    pub(super) event_id_func: fn() -> Uuid,
```

Comment:
If we just allow for V4 | V7 this doesn't even need to be a complex type like a function (pointer/object).

Reply:
I'll change this to store the enum instead (which would also work even with a custom option). It was a leftover from the first approach and then I didn't change it thinking it might be faster to match once and not per event.